config/src/lib.rs

```diff
@@ -904,50 +916,38 @@ impl Config {
     /// This serializes to a table with the name of the profile
     ///
     /// ```toml
-    /// [default]
+    /// [profile.default]
     /// src = "src"
     /// out = "out"
     /// libs = ["lib"]
     /// # ...
     /// ```
     pub fn to_string_pretty(&self) -> Result<String, toml::ser::Error> {
         // serializing to value first to prevent `ValueAfterTable` errors
-        let value = toml::Value::try_from(self)?;
-        let mut s = toml::to_string_pretty(&value)?;
-
-        if self.optimizer_details.is_some() {
-            // this is a hack to make nested tables work because this requires the config's profile
-            s = s
-                .replace("[optimizer_details]", &format!("[{}.optimizer_details]", self.profile))
-                .replace(
-                    "[optimizer_details.yulDetails]",
-                    &format!("[{}.optimizer_details.yulDetails]", self.profile),
-                );
-        }
-        if self.model_checker.is_some() {
-            // similarly to the optimizer details above,
-            // this is a hack to make nested tables work because this requires the config's profile
-            s = ["contracts", "engine", "targets", "timeout"]
-                .iter()
-                .fold(s, |acc, op| {
-                    acc.replace(
-                        &format!("[model_checker.{}]", op),
-                        &format!("[{}.model_checker.{}]", self.profile, op),
-                    )
-                })
-                .replace("[model_checker]", &format!("[{}.model_checker]", self.profile));
-        }
-        s = s.replace("[rpc_storage_caching]", &format!("[{}.rpc_storage_caching]", self.profile));
-
-        if !self.rpc_endpoints.is_empty() {
-            s = s.replace("[rpc_endpoints]", &format!("[{}.rpc_endpoints]", self.profile));
+        let mut value = toml::Value::try_from(self)?;
+        // remove standalone sections from inner table
+        let standalone_sections = Config::STANDALONE_SECTIONS
+            .iter()
+            .filter_map(|section| {
+                let section = section.to_string();
+                value.as_table_mut().unwrap().remove(&section).map(|value| (section, value))
+            })
+            .collect::<Vec<_>>();
```

Comment:
does this need to be executed before the `wrapping_table` is created?

otherwise we can simply used it as iterator for `for (section, value) in ... {`

Reply:
oh, yeah the reason for this is just accessing. the wrapping table takes ownership of the table, and then to access the non-standalone-sections you'd have to have to do this ugly chain of `table.as_table_mut().unwrap().get_mut(PROFILE).unwrap().as_table_mut().unwrap()` etc. But I can change it to the latter if you'd like

Reply:
yikes -.-